dial.go

```diff
@@ -38,16 +39,22 @@ type Dialer struct {
 	// Warning: if PrivateKey is nil, connection will not be secured.
 	PrivateKey ci.PrivKey
 
+	// Protector makes dialer part of a private network.
+	// It includes implementation details how connection are protected.
+	// Can be nil, then dialer is in public network.
+	Protector ipnet.Protector
+
 	// Wrapper to wrap the raw connection (optional)
 	Wrapper WrapFunc
 
 	fallback transport.Dialer
 }
 
-func NewDialer(p peer.ID, pk ci.PrivKey, wrap WrapFunc) *Dialer {
+func NewDialer(p peer.ID, pk ci.PrivKey, pro ipnet.Protector, wrap WrapFunc) *Dialer {
```

Comment:
Don't make this part of the constructor args, just set it on the object returned if we need to:
```go
d := NewDialer(...)
d.Protector = pro
```